src/domain/account/getNfts.js

```diff
@@ -3,11 +3,22 @@ const NFT = require('../../infra/nft');
 
 const nftInstance = new NFT();
 
-async function getNfts({ address, search, chain }) {
-  const nfts = await nftInstance.getOwnedNFTs({
-    address,
-    chain: chain || config.CHAIN,
-  });
+async function getNfts({ address, search, chain, tokenAddress, tokenId }) {
+  let nfts = [];
+
+  if (tokenAddress && tokenId) {
+    nfts = await nftInstance.getNftByMetadata({
```

Comment:
In this case I can paste a URL of any random NFT and it will show up in the modal

Reply:
add check for ownership.